cmd/telliot/main.go

```diff
@@ -132,280 +140,300 @@ func AddDBToCtx(remote bool) error {
-			// Start and wait for it to be ready
-			if err := ds.Start(ctx); err != nil {
-				ExitOnError(err, "starting data server")
-			}
-			<-ds.Ready()
+func (s *stakeCmd) Run(logger log.Logger, client rpc.ETHClient, contract tellorCommon.Contract, account tellorCommon.Account) error {
+	switch s.Operation {
+	case "deposit":
+		return ops.Deposit(ctx, logger, client, contract, account)
+	case "withdraw":
+		return ops.WithdrawStake(ctx, logger, client, contract, account)
+	case "request":
+		return ops.RequestStakingWithdraw(ctx, logger, client, contract, account)
+	case "status":
+		return ops.ShowStatus(ctx, logger, client, contract, account)
+	default:
+		return errors.New("unknown stake command")
+	}
+}
 
-			// Wait for kill sig.
-			<-c
-			// Notify exit channels.
-			ch <- os.Interrupt
-
-			cnt := 0
-			start := time.Now()
-			for {
-				cnt++
-				dsStopped := false
-
-				if ds != nil {
-					dsStopped = !ds.Running
-				} else {
-					dsStopped = true
-				}
-
-				if !dsStopped && cnt > 60 {
-					level.Warn(logger).Log("msg", "taking longer than expected to stop operations", "waited", time.Since(start))
-				} else if dsStopped {
-					break
-				}
-				time.Sleep(500 * time.Millisecond)
-			}
-			level.Info(logger).Log("msg", "main shutdown complete")
+func (b *balanceCmd) Run(client rpc.ETHClient, contract tellorCommon.Contract) error {
+	fmt.Println("balancecmd")
+	addr := ETHAddress{}
+	var err error
+	if b.Address == "" {
+		err = addr.Set(contract.Address.String())
+		if err != nil {
+			return errors.Wrapf(err, "parsing argument")
+		}
+	} else {
+		err = addr.Set(b.Address)
+		if err != nil {
+			return errors.Wrapf(err, "parsing argument")
 		}
+	}
+	return ops.Balance(ctx, client, contract.Getter, addr.addr)
+}
 
+func (c *transferCmd) Run(logger log.Logger, client rpc.ETHClient, contract tellorCommon.Contract, account tellorCommon.Account) error {
+	address := ETHAddress{}
+	err := address.Set(c.Address)
+	if err != nil {
+		return errors.Wrapf(err, "parsing argument")
 	}
+	amount := EthereumInt{}
+	err = amount.Set(c.Amount)
+	if err != nil {
+		return errors.Wrapf(err, "parsing argument")
+	}
+	return ops.Transfer(ctx, logger, client, contract, account, address.addr, amount.Int)
 }
 
-func main() {
-	// Programming is easy. Just create an App() and run it!!!!!
-	app := App()
-	err := app.Run(os.Args)
+func (c *approveCmd) Run(logger log.Logger, client rpc.ETHClient, contract tellorCommon.Contract, account tellorCommon.Account) error {
+	address := ETHAddress{}
+	err := address.Set(c.Address)
 	if err != nil {
-		fmt.Fprintf(os.Stderr, "app.Run failed: %v\n", err)
+		return errors.Wrapf(err, "parsing argument")
 	}
+	amount := EthereumInt{}
+	err = amount.Set(c.Amount)
+	if err != nil {
+		return errors.Wrapf(err, "parsing argument")
+	}
+	return ops.Approve(ctx, logger, client, contract, account, address.addr, amount.Int)
+}
+
+var cli struct {
+	LogConfig logConfigPath `type:"path" help:"path for log config file"`
```

Comment:
for paths to files and dirs best to use https://github.com/alecthomas/kong#custom-named-decoders

`existingfile`, `existingdir`